core/execution/engine.go

```diff
@@ -187,6 +197,17 @@ func NewEngine(
 	return e
 }
 
+// organises common markets copy such that it is always storing all future markets order by their slice then spot markets order by theirs.
```

Comment:
Whats the reason `commonMarketsCpy` it needing to be ordered like this? Given `markets` and `spotMarkets` are ordered by when they were submitted it feels nature for the common one to be like that too.

Reply:
because unless we save this order on its own on the snapshot it would get lost when loading from snapshot.

Reply:
We should save the order in the snapshot then, and not have it dictate the code outside of snapshots. We're just going to expose ourselves to horrible bug where for whatever reason we forget in some place, somewhere, to call the regenerate function and then some snapshot will fail and we'll have to hunt all through the execution engine for it.

Its going to be a pain because we'll have to to an upgrade path thing, but at a minimum something like that guesses an upgrade has happened:
```
if len(commonMarkets) == 0 && len(spotMarkets) == 0 {
    // we've come from an old snapshot, populate common from the future's slice
}
```

or we could be fancy and do it properly and throw a version number into the snapshot and then we can know for sure we've come from an older version.

Reply:
I'm not convinced this is really necessary given that there are exactly 2 places where regenerate is called (new market and restore market). I also feel like there could be functional reasons to process one set of markets before the others but there is currently no such requirement.

Reply:
The reasons are:
1) having the snapshot rely on an assumption always goes wrong
2) there is benefit to keeping the ordering of _all_ the slices consistent, namely, ordered by when they were submitted
3) we're letting awkwardness of snapshots dictate code outside of snapshots, to cut a corner and make things easier _right now_

Reply:
I think it all comes down to your assumption that the order of iteration must reflect the order of insertion, to which I don't see a compelling reason to so far. 

1. I don't think saving duplicate information in the snapshot is a good idea. One can argue that this is not duplicate if the creation order across types *must* be maintained, but I haven't seen an argument for why it must yet.
2. what is this benefit?
3. It's true that if there wasn't for the snapshot we'd just have it in the insertion order because it's the easy thing, but i don't see how rearranging it by market type - then insertion order is cutting a corner or making things easier _right now_. 

anyways, I concede, I'll add it to the snapshot.